src/app/clusters/bindings/bindings.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Binding Server Cluster
+ ***************************************************************************/
+
+#include "af.h"
+
+#include <app/util/binding-table.h>
+#include <support/logging/CHIPLogging.h>
+
+using namespace chip;
+
+EmberStatus prepareBinding(EmberBindingTableEntry & binding, NodeId nodeId, GroupId groupId, EndpointId endpointId,
+                           ClusterId clusterId)
+{
+    if (groupId && nodeId)
+    {
+        return EMBER_BAD_ARGUMENT;
+    }
+
+    if (groupId)
+    {
+        binding.type      = EMBER_MULTICAST_BINDING;
+        binding.groupId   = groupId;
+        binding.clusterId = clusterId;
+        binding.local     = emberAfCurrentCommand()->apsFrame->destinationEndpoint;
+    }
+    else
+    {
+        binding.type      = EMBER_UNICAST_BINDING;
+        binding.nodeId    = nodeId;
+        binding.clusterId = clusterId;
+        binding.local     = emberAfCurrentCommand()->apsFrame->destinationEndpoint;
+        binding.remote    = endpointId;
+    }
+
+    return EMBER_SUCCESS;
+}
+
+EmberStatus getBindingIndex(EmberBindingTableEntry & newEntry, uint8_t * bindingIndex)
+{
+    EmberBindingTableEntry currentEntry;
+    for (uint8_t i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+    {
+        emberGetBinding(i, &currentEntry);
+        if (currentEntry.type == EMBER_UNUSED_BINDING)
+        {
+            continue;
+        }
+
+        if (currentEntry.type == newEntry.type && currentEntry.clusterId == newEntry.clusterId &&
```

Comment:
some operator== or a helper method could make the code more readable.

Reply:
Done.